api/src/controllers/login.js

```diff
@@ -51,6 +62,31 @@ const templates = {
       'privacy.policy'
     ],
   },
+  passwordReset: {
+    file: path.join(__dirname, '..', 'templates', 'login', 'password-reset.html'),
+    translationStrings: [
+      'login.show_password',
+      'login.hide_password',
+      'change.password.title',
+      'change.password.hint',
+      'change.password.submit',
+      'change.password.new.password',
+      'change.password.confirm.password',
+      'password.weak',
+      'password.length.minimum',
+      'password.must.match',
+      'user.password.current',
+      'password.current.incorrect',
+      'password.same'
+    ],
+  }
+};
+
+const skipPasswordChange = async (user, userCtx) => {
+  return !user?.password_change_required ||
+    user?.token_login?.active ||
+    roles.isDbAdmin(userCtx) ||
+    await auth.hasAllPermissions(userCtx, 'can_skip_password_change');
```

Comment:
Thanks for catching this. I now see the disconnect between when we set the requirement and when we check if it can be skipped. Let me move the logic to users.js then in login controller just check value of `password_change_required`.